src/component/reducer/actions/SpectrumsActions.ts

```diff
@@ -116,6 +116,7 @@ function handleChangeActiveSpectrum(draft: Draft<State>, activeSpectrum) {
     }
     activeSpectrum = { ...activeSpectrum, index: newIndex };
     draft.activeSpectrum = activeSpectrum;
+    draft.view.spectra.currentSpectrumId = activeSpectrum.id;
     draft.tabActiveSpectrum[draft.activeTab] = activeSpectrum;
```

Comment:
Using the activeTab is important because for each tab we have an active spectrum. So I was wrong in my issue.

Currently it surprisingly works but I don't understand why ... This `spectra` property should contain a property `activeSpectra` that is an object that contains one property per tab. So all the changes must depend on `activeTab`

Reply:
in the project we use `tabActiveSpectrum` to save the active spectrum for each tab, so
do you want me to add tab active spectrum and active tab in the view?

Reply:
Thanks for checking this. Then I don't know where is currentSpectrumId (is it used ?)

I would expect that 'tabActiveSpectrum' goes indeed in

view.spectra.activeSpectraId[activeTab]

We should check if we need to keep the activeSpectrum or just it's id. This may require a lot of changes so we will currently wait with this PR because there is another one (load-save) that changes a lot of code.

Reply:
we need the index in `hasAcceptedSpectrum` function in `ToolsActions`